manager/src/main/java/org/openremote/manager/mqtt/TeltonikaMQTTHandler.java

```diff
@@ -0,0 +1,627 @@
+     * Should generally check the third token (index 2) onwards unless {@link #handlesTopic} has been overridden.
+     *
+     */
+    @Override
+    protected boolean topicMatches(Topic topic) {
+        return TELTONIKA_DEVICE_TOKEN.equalsIgnoreCase(topicTokenIndexToString(topic, 2));
+    }
+
+    @Override
+    public void start(Container container) throws Exception {
+        super.start(container);
+        getLogger().info("Starting Teltonika MQTT Handler");
+        ManagerIdentityService identityService = container.getService(ManagerIdentityService.class);
+        assetStorageService = container.getService(AssetStorageService.class);
+        AssetDatapointService = container.getService(AssetDatapointService.class);
+        timerService = container.getService(TimerService.class);
+        DeviceParameterPath = Paths.get("deployment/manager/fleet/FMC003.json");
+        if (!identityService.isKeycloakEnabled()) {
+            getLogger().warning("MQTT connections are supported when not using Keycloak identity provider, only for the Teltonika Telematics devices, until auto-provisioning is fully implemented.");
+            isKeycloak = false;
+        } else {
+            isKeycloak = true;
+            identityProvider = (ManagerKeycloakIdentityProvider) identityService.getIdentityProvider();
+        }
+        //TODO: Figure out a way to update the AssetFilter when a new asset is a candidate to receive a response message
+        // Does it even automatically refresh the Asset list?
+
+        clientEventService.addInternalSubscription(
+                AttributeEvent.class,
+                buildAssetFilter(),
+                this::handleAttributeMessage);
+    }
+
+    /**
+     * Creates a filter for the AttributeEvents that could send a command to a Teltonika Device.
+     *
+     * @return AssetFilter of CarAssets that have both {@value TELTONIKA_DEVICE_RECEIVE_COMMAND_ATTRIBUTE_NAME} and
+     * {@value TELTONIKA_DEVICE_SEND_COMMAND_ATTRIBUTE_NAME} as attributes.
+     */
+    private AssetFilter<AttributeEvent> buildAssetFilter(){
+        List<Asset<?>> assetsWithAttribute = assetStorageService
+                .findAll(new AssetQuery().types(CarAsset.class)
+                .attributeNames(TELTONIKA_DEVICE_SEND_COMMAND_ATTRIBUTE_NAME));
+        List<String> listOfCarAssetIds = assetsWithAttribute.stream()
+                .map(Asset::getId)
+                .toList();
+
+        AssetFilter<AttributeEvent> event = new AssetFilter<>();
+        event.setAssetIds(listOfCarAssetIds.toArray(new String[0]));
+        event.setAttributeNames(TELTONIKA_DEVICE_SEND_COMMAND_ATTRIBUTE_NAME);
+        return event;
+    }
+
+    private void handleAttributeMessage(AttributeEvent event) {
+        // If this is not an AttributeEvent that updates a TELTONIKA_DEVICE_SEND_COMMAND_ATTRIBUTE_NAME field, ignore
+        if (!Objects.equals(event.getAttributeName(), TELTONIKA_DEVICE_SEND_COMMAND_ATTRIBUTE_NAME)) return;
```

Comment:
Check asset type of event also

Reply:
https://github.com/openremote/openremote/pull/1147#discussion_r1401988178